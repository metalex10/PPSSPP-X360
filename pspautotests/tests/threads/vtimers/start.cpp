#include "shared.h"

int countHits = 0;
SceUInt countHandler(SceUID uid, SceKernelSysClock *scheduled, SceKernelSysClock *actual, void *common) {
	++countHits;
	return 0;
}

extern "C" int main(int argc, char *argv[]) {
	SceUID vtimer = sceKernelCreateVTimer("test", NULL);

	checkpointNext("Objects");
	checkpoint("  Normal: %08x", sceKernelStartVTimer(vtimer));
	checkpoint("  Twice: %08x", sceKernelStartVTimer(vtimer));
	checkpoint("  NULL: %08x", sceKernelStartVTimer(0));
	checkpoint("  Invalid: %08x", sceKernelStartVTimer(0xDEADBEEF));
	sceKernelDeleteVTimer(vtimer);
	checkpoint("  Deleted: %08x", sceKernelStartVTimer(vtimer));

	vtimer = sceKernelCreateVTimer("test", NULL);
	checkpointNext("While scheduled");
	sceKernelSetVTimerTimeWide(vtimer, 10000);
	SceKernelSysClock t = {100, 0};
	sceKernelSetVTimerHandler(vtimer, &t, &countHandler, (void *)0x00001337);
	sceKernelStartVTimer(vtimer);
	sceKernelDelayThread(1000);
	sceKernelStopVTimer(vtimer);
	checkpoint("  After start: %d", countHits);

	sceKernelDeleteVTimer(vtimer);
	return 0;
}