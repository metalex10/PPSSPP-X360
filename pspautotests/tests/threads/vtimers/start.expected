[x] Objects
[x]   Normal: 00000000
[x]   Twice: 00000001
[x]   NULL: 800201bf
[x]   Invalid: 800201be
[x]   Deleted: 800201be

[x] While scheduled
[r]   After start: 1
